### DISCUSSION TITLE
Which part of the repo actually download valheim the game?

### DISCUSSION THREAD

**Discussion by dragondaniel** [lloesche/valheim-server-docker#64] [Q&A] [answered]
Hi,

Wanted to learn, I understand the part the steamcmd is being installed.
But looking around, I can't seem to find the exact part which is actually installing valheim game server.

Cause steamcmd, is just installing steam and then login as anonymous.

Could someone please help shed some light of where the actual server in being installed?

**Comment by anpross** [comment]
not the author but the way i see it, the Dockerfile is downloading the "Steam CLI" here: https://github.com/lloesche/valheim-server-docker/blob/main/Dockerfile#L5 which is executed below and downloads itself and gets backed-in into the docker image.

In the server-start script here, https://github.com/lloesche/valheim-server-docker/blob/main/valheim-server#L13 there is an export of the SteamAppId which tells the generic Steam server what server to run.

So the Steam platform can distribute the server code the same way it is distributing the client.

**Comment by lloesche** [comment, marked as answer]
This is what downloads the server on startup and also periodically checks for updates: https://github.com/lloesche/valheim-server-docker/blob/0e2ef3bc2ca9f53f21a9f0538239c63b38db0349/valheim-updater#L27

Downloading the dedicated server does not require a Steam account.

**Reply by dragondaniel** [reply]
Thank you very much, this helps alot.
